src/less/skeleton/skeleton.less

```
@@ -116,11 +116,11 @@ span.skeleton__textbox + span {
     margin-inline-start: var(--spacing-100);
 }
 
-div.skeleton__avatar,
-div.skeleton__button,
-div.skeleton__text,
-div.skeleton__image,
-div.skeleton__textbox {
+div.skeleton__avatar + div,
```

This seems a little too general as a selector since it applies to _all_ divs, would something like this work?

```suggestion
div.skeleton__avatar:not(:first-child),
```

This also applies to the ` + span` selectors in the block above this one.

---

@agliga Is in charge of branch health. I agree 17.1.0 needs keeping up to date daily with master

---

Inline elements have (8px) different spacing requirements than block (10px) and hence needed that clear distinction.

---

Hmm...why do we care about what the inside element spacing looks like? Shouldn't the spacing of elements inside just stay as it is?

---

We need some sensible default for vertical margins between skeletons pieces where possible, otherwise all app teams will end up having to figure out what that spacing is and write the css themselves.

---

Gotcha. That makes sense. Sorry, for some reason, I misread the `+` as `>`.

Would @LuLaValva 's concern of other `div` elements getting selected be technically possible?

---

Yes, this would effect any `div` elements that devs use for their custom skeletons. We could also set `margin-after` instead of `margin-before` and change the selector to `* + skeleton__[]` or `skeleton__[]:not(:last-child)`

---

It's probably preferable to be explicit with classes. If I'm understanding what we're trying to select, I believe this might work:

```css
:where(.skeleton) 
:where(
    div.skeleton__avatar, 
    div.skeleton__button, 
    div.skeleton__text, 
    div.skeleton__image, 
    div.skeleton__textbox
    ) {
    margin-block-start: var(--spacing-150);
}
```

---

> Yes, this would effect any `div` elements that devs use for their custom skeletons.

If you are talking about piecing together our avatar, image, text, textbox etc, well that was kind of the point. We want those div elements to have a sensible default. 

If you are talking about fully custom skeletons (i.e. nothing in that list that we offer), then I think having our sensible default is an okay place to start and they are free to override it (given that they will be needing to write some css anyway)

---

Maybe we're overthinking this. 😆 What @saiponnada has + @LuLaValva 's suggestion to avoid first child might be the way to go?

---

I kind of agree, it's difficult to cover all these use cases.  `div.skeleton__avatar:not(:first-child)` improves coverage than the current.

---

Changed to `.skeleton__avatar:not(:last-child)` along with `margin-block-end`.